xlators/features/index/src/index.c

```diff
@@ -1756,7 +1756,8 @@ index_readdir_wrapper(call_frame_t *frame, xlator_t *this, fd_t *fd,
         goto done;
     }
 
-    dir = fctx->dir;
+    if (fctx)
```

Comment:
I agree that ret !>0  check at 1752  by clang is FP as ret cannot be greater than 0 and if ret = 0, fctx will not be NULL